### DISCUSSION TITLE
[Feature] Access albums shared via link in mobile app

### DISCUSSION THREAD

**Discussion by thielepaul** [immich-app/immich#2327] [Feature Request]
### The feature

My usecase for this feature is the following: After a trip with friends, everyone wants to have access to the photos from the others.
To achieve this, I would like to do the following:
- I create an album on my immich server
- I create a sharing link and allow uploading for public users
- I send an appstore link to immich and the album link to my friends
- they download immich and instead of logging in they can paste the link and thus get access to the album and upload their folder
- if someone is already logged in to another immich server, it should be possible for him to add the sharing link via a menu option

I cannot use the web app for this usecase, because metadata is stripped from photos on ios devices. Furthermore, if I upload many photos, the app should be able to continue if the upload is interrupted, while I would need to restart the process in the browser.

Is this something that is in the scope for immich?

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by alextran1502** [comment]
There is a plan to integrate public view into the mobile app. Though we are not sure how to carry out that yet, your suggestion can be a potential implementation :) Thank you

**Comment by rtisma** [comment]
ufff. This seems very non-trivial, but i think this has huge value and potential. 

I had an event this summer and wanted family members to send their photos. Given that in these situations most people have photos on a smart phone, upload from a desktop is not as popular. In addition, mobile uploads via browser are not practical for the reasons above. A feature like this would useful because it would help eliminate temp user creation and provide a more reliable solution to uploading media. 

I think the typical steps would be:
     - user downloads immich mobile app
     - admin generates an immich share link with expiry
     - admin create a user and permissions
     - admin sends user their login creds
     - admin sends immich link with expiry
     - user upload photos
     - admin waits for expiry to pass and disables/deletes user.  
     
The problematic steps are temp user creation and deletion. These can usually be a turn off for people that just want to upload photos without signing up.  Again, no user creation needed to desktop/browser uploads, but from a mobile phone, uploading via the browser is just not practical.

  With this feature request the steps would be:
     - user download immich mobile app
     - admin generates an immich share link with expiry
     - admin sends immich link with expiry to user
     - user upload photos
      
 As an admin, i don't need to track the expiry to know when to delete/disable the user from access to immich (if any) and can just rely on the expiry of the link. 
 
 As a user, I don't need to login and can just upload anonymously thus providing a quick and simple experience which motivates users to actually upload.

**Comment by FreeWind6** [comment]
Any news when this feature is expected?)

**Comment by diogotcorreia** [comment]
I want to add that this is quite important for my friends with iOS devices, because Apple does not let users upload the video part of live photos on a browser, and they end up uploading a still photo. Also, it appears the file picker is also very laggy for them :/

The workaround I've found at the moment is for them to send me an iCloud share link, which includes everything with full resolution, but only works if they are using iCloud to backup their photos. I can then upload that to Immich no problem.

**Reply by diogotcorreia** [reply]
In addition to the ones in the OP, some more design considerations off the top of my head:

- Recently seen share links should be saved, so you can keep accessing them without getting the link again
- It should be possible to open a share link from an Immich instance other than the one you are signed in to
- It's not reasonable for share links to open directly in the app because it's not possible to know _all_ the Immich instances out there. It might be interesting to add a new endpoint to `https://my.immich.app/` for this, something like `https://my.immich.app/share#https://immich.example.com/share/...` (using `#` so that the link isn't sent to the backend for privacy reasons, not sure if it can be forwarded to the app properly :/)
  - Another option could be for the web app on mobile to have a link somewhere to open it on the app

I unfortunately don't know much about mobile development, but I'd be happy to test all of this if/when it is implemented, so feel free to ping me if/when a PR is open.
Keep up the awesome work! :heart:

**Comment by ravishekharco** [comment]
+1
This is now the only thing stopping me from switching over to Immich completely. Holding off importing my 800k+ photos waiting for this.

**Comment by joelarm** [comment]


**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by Ecuashungo** [comment]
I would really like to see this feature as well. I've just recently started to use immich and its amazing, thanks to everyone!

Is there anything I can do to help move this feature request forward? With the current architecture, would this even be possible to implement from a technical standpoint?
